### PR TITLE
Creating List

### PR DESCRIPTION
**Pull request by lihs-ie**
## 📌 What does this change do?

Implements an `ImmutableList<T>` utility that provides a read-only, value-oriented interface over a list.  
All mutating operations return a new list instance, preserving immutability.

---

## 🧠 Why is this change necessary?

- Avoids accidental side effects from direct array mutation
- Enables functional programming patterns by returning new instances on operations
- Improves predictability and testability in state management and data transformation
- Offers a safer alternative to native `Array<T>` when immutability is preferred

---

## 🛠️ How was it implemented?

- Wraps a standard `T[]` internally, but exposes only immutable methods
- Operations like `map`, `filter`, `append`, `removeAt`, etc. return new `ImmutableList<T>` instances
- Deep equality for `.equals()` comparison
- Implements `toArray()` for interoperability
- Includes unit tests for all methods, including edge cases (e.g., empty list, out-of-bounds removal)

---

## ✅ Checklist

- [x] Code is formatted (Prettier, etc.)
- [x] ESLint passes without errors
- [x] All relevant tests pass
- [x] Unit tests added for all public methods
- [x] No breaking changes introduced
- [x] JSDoc or inline documentation written
- [x] `ImmutableList` behaves as a pure, side-effect-free structure

---

## 💬 Notes for future me

- Consider adding `.flatMap()` in a future PR
- Benchmark performance on large lists if used in hot paths

### CODE REVIEW

File: src/collections/list/common.ts [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+    callback?: (left: T, right: T) => boolean,
+  ) => boolean;
+  exists: (predicate: (value: T) => boolean) => boolean;
+  forall: (predicate: (value: T) => boolean) => boolean;
+  foldLeft: <R>(initial: R, callback: (accumulator: R, value: T) => R) => R;
+  foldRight: <R>(initial: R, callback: (accumulator: R, value: T) => R) => R;
+}
+
+export const ImmutableList = <T>(values: T[] = []): ImmutableList<T> => {
+  const items = values;
```
[the remark this answers is not in the document]
Comment (lihs-ie) [reply src/collections/list/common.ts]: Thank you for reviewing

File: src/collections/list/common.ts [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+  equals: (
+    comparison: ImmutableList<T>,
+    callback?: (left: T, right: T) => boolean,
+  ) => boolean;
+  exists: (predicate: (value: T) => boolean) => boolean;
+  forall: (predicate: (value: T) => boolean) => boolean;
+  foldLeft: <R>(initial: R, callback: (accumulator: R, value: T) => R) => R;
+  foldRight: <R>(initial: R, callback: (accumulator: R, value: T) => R) => R;
+}
+
```
[the remark this answers is not in the document]
Comment (lihs-ie) [reply src/collections/list/common.ts:33]: I have a policy not to write JSDoc.
